File: Semester 2/OOP/a45-913-Fintina-Olivia/testController.cpp
#include <assert.h>
#include "Controller.h"
#include "TestAll.h"

void testController()
{
    std :: cout << "Testing controller..." << std :: endl;
    Repository repository;
    Controller controller(repository);
    controller.setFileName("trenchCoats.txt");
    //std :: cout << "First test passed!" << std :: endl;

    controller.addTrenchCoat(1, 2, "red", 3, 4, "photo");
    assert(controller.getSize() == 1);
    assert(controller.trenchCoatExists(1) == true);
    //std :: cout << "Second test passed!" << std :: endl;

    controller.removeTrenchCoat(1);
    assert(controller.getSize() == 0);
    //std :: cout << "Third test passed!" << std :: endl;

    controller.addTrenchCoat(1, 2, "red", 3, 4, "photo");
    assert(controller.getTrenchCoat(1).getTrenchColour() == "red");
    controller.updateTrenchCoat(1, 3, 2, "blue", 3, 4, "photo");
    assert(controller.getTrenchCoat(3).getTrenchColour() == "blue");
    //std :: cout << "Fourth test passed!" << std :: endl;

    controller.updateTrenchCoat(3, 3, 2, "blue", 3, 4, "photo");
    assert(controller.getSize() == 1);
    assert(controller.updateTrenchCoat(5, 3, 2, "blue", 3, 4, "photo") == false);



    assert(controller.getSize() == 1);
    assert(controller.getFileName() == "trenchCoats.txt");
}

void testTrenchCoatExists()
{
    Repository repository;
    Controller controller(repository);
    controller.setFileName("trenchCoats.txt");

    controller.addTrenchCoat(1, 2, "red", 3, 4, "photo");
    assert(controller.trenchCoatExists(1) == true);
    assert(controller.trenchCoatExists(2) == false);

    std :: cout << "Controller tests passed!" << std::endl;
}

File: Semester 3/Advanced Programming Methods (MAP)/Assignment 1/src/tests/TestController.java
package tests;

import controller.Controller;
import module.Sphere;
import repository.Repository;
import module.Cube;

public class TestController {
    public static void test(){
        Repository repository = new Repository();
        Controller controller = new Controller(repository);

        controller.addItem(new Cube(5));
        controller.addItem(new Cube(10));
        controller.addItem(new Sphere(5));

        try {
            controller.addItem(new Sphere(-5));
            System.out.println("Test AddItem: Passed (test)");
        } catch (Exception e) {
            System.out.println("Test AddItem: Failed (test)");
        }

//        int length = controller.getCurrentSize();
//        System.out.println("Length " + length);
//        for(int i = 0; i < length; i++){
//            System.out.println(controller.getAll().get(i).getVolume());
//        }

        if(controller.filter(150).size() != 2){
            System.out.println("Test Filter: Failed");
        }
        else{
            System.out.println("Test Filter: Passed");
        }
    }

}
